extensions/datastores/accumulo/src/main/java/mil/nga/giat/geowave/datastore/accumulo/query/WholeRowAggregationIterator.java

```diff
@@ -0,0 +1,183 @@
+package mil.nga.giat.geowave.datastore.accumulo.query;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.hadoop.io.Text;
+
+import mil.nga.giat.geowave.core.store.data.CommonIndexedPersistenceEncoding;
+import mil.nga.giat.geowave.core.store.data.PersistentDataset;
+import mil.nga.giat.geowave.core.store.index.CommonIndexValue;
+import mil.nga.giat.geowave.datastore.accumulo.encoding.AccumuloFieldInfo;
+
+public class WholeRowAggregationIterator extends
+		WholeRowQueryFilterIterator
+{
+	private AggregationIterator aggregationIterator;
+
+	public WholeRowAggregationIterator() {
+		super();
+	}
+
+	@Override
+	protected boolean filter(
+			final Text currentRow,
+			final List<Key> keys,
+			final List<Value> values ) {
+		if ((aggregationIterator != null) && (aggregationIterator.queryFilterIterator != null)
+				&& aggregationIterator.queryFilterIterator.isSet()) {
+			final PersistentDataset<CommonIndexValue> commonData = new PersistentDataset<CommonIndexValue>();
+			final List<AccumuloFieldInfo> unknownData = new ArrayList<AccumuloFieldInfo>();
```

Comment:
This is a bit odd and deserves an explanation.
The check applies to the aggregationIterator and its queryFilterIterator.
The code below uses 'this' class instance queryFilterIterator